### DISCUSSION TITLE
Ignoring spurious temp databases

### DISCUSSION THREAD

**Discussion by UnixRonin** [mydumper/mydumper#1219] [Q&A]
I'm using mydumper, in a wrapper script, to perform nightly backups on a three-node MariaDB 10.6.14 Galera cluster on Gentoo Linux.  All nodes of a Galera cluster are supposed to have the same contents.  However, I am seeing periodic dump failures because **one node only** creates — and never cleans up — an empty temp database with a filename like **tmp.FG70vxrhiH**, which appears in MySQL/MariaDB as **#mysql50#tmp.FG70vxrhiH**.  I have yet to successfully figure out an exclusion regexp to tell it to ignore all such "databases".

Can anyone help me out here?

**Comment by davidducos** [comment]
I'm not an expert on regex but if you follow this: https://github.com/mydumper/mydumper#how-to-exclude-or-include-databases you can write a rule like this:  `--regex '^(?!(FG70vxrhiH\.))'` which, if I'm not wrong, will filter everything in a schema that ends with `FG70vxrhiH`.

**Reply by UnixRonin** [reply]
Yes, but that particular form doesn't help me because that's the random-variable part of the name.  What I need to match on is the #mysql50# part, and I haven't figured out any way to do that.   I also tried to match on the 'tmp.' substring, since it's the embedded period that should trigger the #mysql50# internal prefix, and haven't been able so far to get that to work either.  I suspect I haven't gotten the escaping syntax right.

Of course what I'd REALLY like would be to figure out what keeps creating these weird empty temp DBs on just one single node and not cleaning them up, and kill it with fire.

**Reply by davidducos** [reply]
What about  `--regex '^(?!(.*mysql50.*\.))'` ? does it work for you?

**Reply by UnixRonin** [reply]
Nope, that form fails as well.

** (mydumper:15591): CRITICAL **: 10:33:32.549: Error showing events on: #mysql50#tmp.test - Could not execute query: Incorrect database name '#mysql50#tmp.test'

**Reply by davidducos** [reply]
@UnixRonin, I reviewed the issue, and there is nothing that we can do, is this is not a real schema, that is something that you need to fix at db configuration level. 
The root cause of the issue is that mysql trust on the directories on the data dir, instead of the data dictionary. This has pros and cons that we are not going to discuss/solve here. 
Sorry that we can't help you.

**Reply by UnixRonin** [reply]
_*nod*_  totaslly understand.  My primary goal is to try to figure out what's creating the spurious directories there, so that I can Make Them Go Away, but I was hoping in the meantime to find a way to at least make mydumper ignore them until I can identify and fix the underlying issue.
